File: ESAResturant.NEW.web/ClientApp/src/app/admin/bill-invoice/Models/BillDetail.ts
import { Item } from '../../item/Models/Item';
import { ItemPrice } from '../../item/Models/itemPrice';
import { Unit } from '../../item/Models/Unit';
import { Bill } from './Bill';

export class BillDetail {

  constructor() {

    this.id = 0;
    this.billId = 0;
    this.itemId = 0;
    this.itemSellPrice = 0;
    this.itemBuyPrice = 0;
    this.qty = 1;
    this.unitId = 0;
    this.smallUnitQty = 0;

    this.supTotal = 0;
    this.vatTax = 0;
    this.vatTaxValue = 0;
    this.totalAfterVatTax = 0;

    this.discount = 0;
    this.discountValue = 0;

    this.notePrice = 0;
    this.netTotal = 0;
    this.itemPriceId = 0;

    this.currentDate = new Date();
    this.date = new Date();
    this.tableNo = '';
    this.note = '';
    this.isNew = false;
    this.isDelete = false;
    this.isFire = false;
    this.isSeparate = false;
    this.isFinshed = false;

    this.bill = new Bill();
    this.unit = new Unit();
    this.item = new Item();
    this.itemPrice = new ItemPrice();
  }





  id: number;
  billId: number;
  itemId: number;
  itemSellPrice: number;
  itemBuyPrice: number;
  qty: number;
  unitId: number;
  smallUnitQty: number;
  supTotal: number;    //total after vat calac 

  vatTax: number;
  vatTaxValue: number;
  totalAfterVatTax: number;
  discount: number;
  discountValue: number;
  netTotal: number;
  currentDate: Date;
  date: Date;
  notePrice: number;
  note: string;
  tableNo: string;
  itemPriceId: number;
  isNew?: boolean;
  isDelete?: boolean;

  isPrint?: boolean;
  isFire?: boolean;
  isSeparate?: boolean;
  isFinshed?: boolean;
  bill:Bill
  item: Item;
  unit: Unit;
  itemPrice: ItemPrice;
}

File: ESC.Resturant.Data/Entities/BillDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ESC.Resturant.Data.Entities
{
    public class BillDetails
    {

        [Key]
        public int Id { get; set; }
        public Nullable<int> BillId { get; set; }
        public Nullable<int> ItemId { get; set; }
        public Nullable<double> ItemSellPrice { get; set; }
        public Nullable<double> ItemBuyPrice { get; set; }
        public Nullable<double> Qty { get; set; }
        public Nullable<int> UnitId { get; set; }
        public Nullable<double> SmallUnitQty { get; set; }
        public Nullable<double> SupTotal { get; set; }
        public Nullable<double> VATTax { get; set; }
        public Nullable<double> VatTaxValue { get; set; }
        public Nullable<double> TotalAfterVatTax { get; set; }
        public Nullable<double> Discount { get; set; }
        public Nullable<double> DiscountValue { get; set; }
        public Nullable<double> NetTotal { get; set; }
        public Nullable<System.DateTime> CurrentDate { get; set; }
        public Nullable<System.DateTime> Date { get; set; }
        public string TableNo { get; set; }
        public string Note { get; set; }
        public Nullable<double> NotePrice { get; set; }
        public Nullable<int> ItemPriceId { get; set; }
        public Nullable<bool> IsNew { get; set; }
        public Nullable<bool> IsFinshed { get; set; }
        public Nullable<bool> IsDelete { get; set; }


        public virtual Unit Unit { get; set; }
        public virtual Bill Bill { get; set; }
        public virtual Item Item { get; set; }
        public virtual ItemPrices ItemPrice { get; set; }

    }
}
